trieste/acquisition/function.py

```diff
@@ -745,9 +793,20 @@ def prepare_acquisition_function(
         feasible_query_points = tf.boolean_mask(objective_dataset.query_points, is_feasible)
         feasible_mean, _ = objective_model.predict(feasible_query_points)
         eta = tf.reduce_min(feasible_mean, axis=0)
-        function = expected_improvement(objective_model, eta)
+        tf.debugging.Assert(isinstance(self._improvement_fn, expected_improvement), [])
+        self._improvement_fn.update(eta)  # type: ignore
```

Comment:
Is the expectation that if lines 747-748 are entered (early return in `prepare`), execution never reaches this line here? If it does, this assertion will fail

Reply:
Good catch. This does need to be handled (and I'll make sure it's covered in a unit test).